test/test_jit.py

```diff
@@ -1222,8 +1222,9 @@ def backward(self, grad_output):
                 return grad_output
 
         x = torch.tensor([0.], requires_grad=True)
-        with self.assertRaisesRegex(RuntimeError, "MyLegacyFn"):
-            torch.jit.get_trace_graph(lambda x: MyLegacyFn()(x), (x,))
+        with warnings.catch_warnings(record=True):
```

Comment:
there's no point of `record=True` if you don't use the returned `warnings.catch_warnings` value anywhere

Reply:
They still get printed if `record=False`, https://docs.python.org/3/library/warnings.html#warnings.catch_warnings